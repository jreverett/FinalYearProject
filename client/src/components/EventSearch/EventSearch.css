#event-search-container {
  display: flex;
  align-items: center;
}

#event-search-label {
  position: relative;
  width: 380px;
  height: 70px;
  margin-bottom: 10px;
  display: flex;
  align-items: center;
  justify-content: flex-start;
  text-align: center;
}

.event-search-text {
  font-size: 375%;
}

#event-search-label > span,
#event-search-label > input {
  position: absolute;
}

#event-search-input {
  height: 80px;
  width: 380px;
  border: none;
  background-color: transparent;
  outline: 0;
  border-bottom: 1px solid rgba(110, 110, 110, 0.3);
}

#event-search-input-overlay {
  pointer-events: none;
  white-space: nowrap;
  overflow: scroll;
}
