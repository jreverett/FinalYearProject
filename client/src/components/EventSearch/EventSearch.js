import React, { Component } from 'react';
import Typist from 'react-typist';
import { getRandomNoRepeats } from '../../utilities';
import '../../../node_modules/react-typist/dist/Typist.css';
import '../../common.css';
import './EventSearch.css';

class EventSearch extends Component {
  constructor(props) {
    super(props);

    this.state = {
      searchValue: '',
      searchFocused: false,
    };
  }

  handleSearchChange = (e) => {
    this.setState({ searchValue: e.target.value });
  };

  handleSearchFocus = () => {
    this.setState({ searchFocused: true });
  };

  handleSearchBlur = () => {
    this.setState({ searchFocused: false });
  };

  render() {
    const { searchValue, searchFocused } = this.state;

    const exampleSearches = [
      'Pub Quiz',
      'Music Event',
      'Game Night',
      'Dance Class',
      'Art Class',
      'Movie Night',
      'Football Match',
    ];

    var selectSearch = getRandomNoRepeats(exampleSearches);
    var searches = [];

    for (let i = 0; i < exampleSearches.length; i++) {
      const search = selectSearch();
      searches.push(search);
      searches.push(<Typist.Backspace count={search.length} delay={2000} />);
    }

    return (
      <div id="event-search-container">
        <label id="event-search-label">
          <input
            id="event-search-input"
            className="event-search-text"
            value={searchValue}
            onChange={this.handleSearchChange}
            onFocus={this.handleSearchFocus}
            onBlur={this.handleSearchBlur}
          />

          {/* Overlay should only be visible before the textboxt is focused */}
          {!searchFocused && (
            <span id="event-search-input-overlay" className="event-search-text">
              <Typist
                cursor={{
                  hideWhenDone: true,
                }}
              >
                {searches}
              </Typist>
            </span>
          )}
        </label>
      </div>
    );
  }
}

export default EventSearch;
