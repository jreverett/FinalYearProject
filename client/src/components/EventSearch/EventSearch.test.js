import React from 'react';
import renderer from 'react-test-renderer';
import EventSearch from './EventSearch';

describe('<EventSearch />', () => {
  it('matches the snapshot', () => {
    const tree = renderer.create(<EventSearch />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
