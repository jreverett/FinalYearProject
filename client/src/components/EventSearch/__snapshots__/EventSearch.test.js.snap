// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EventSearch /> matches the snapshot 1`] = `
<form
  id="event-search-container"
  onSubmit={[Function]}
>
  <label
    id="event-search-label"
  >
    <input
      className="event-search-text"
      id="event-search-input"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      value=""
    />
    <span
      className="event-search-text"
      id="event-search-input-overlay"
    >
      <div
        className="Typist "
      >
        <span
          className="Cursor Cursor--blinking"
        >
          |
        </span>
      </div>
    </span>
  </label>
</form>
`;
